### PR TITLE
Doc merge_analysis and some enums

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

File: base/libgit2/consts.jl [hunk 1]
```diff
@@ -160,11 +160,29 @@ module Consts
...
+    * `MERGE_PREFERENCE_NO_FASTFORWARD`: do not allow any fast-forward merges.
+    * `MERGE_PREFERENCE_FASTFORWARD_ONLY`: allow only fast-forward merges and no
+      other type (which may introduce conflicts).
+    """
     @enum(GIT_MERGE_PREFERENCE, MERGE_PREFERENCE_NONE             = 0,
                                 MERGE_PREFERENCE_NO_FASTFORWARD   = 1,
                                 MERGE_PREFERENCE_FASTFORWARD_ONLY = 2)
-
+    """ Result of analysis on merge possibilities.
+    * `MERGE_ANALYSIS_NONE`: it is not possible to merge the elements of `anns`.
```
Comment (ararslan) [inline base/libgit2/consts.jl]: What does `anns` refer to here?

File: doc/src/devdocs/libgit2.md [hunk 2]
```diff
@@ -93,6 +93,7 @@ Base.LibGit2.lookup_branch
 Base.LibGit2.mirror_callback
 Base.LibGit2.mirror_cb
 Base.LibGit2.message
+Base.LibGit2.merge_analysis
```
Comment (ararslan) [inline doc/src/devdocs/libgit2.md]: Maybe add the newly documented enums in here too?

Comment (kshyatt) [reply doc/src/devdocs/libgit2.md]: I don't think we have the other enums in this...

Comment (ararslan) [reply doc/src/devdocs/libgit2.md]: Seems like we should though, at least IMO.

Comment (kshyatt) [reply doc/src/devdocs/libgit2.md]: See [here](https://github.com/JuliaLang/julia/pull/23420#discussion_r135419373) for a discussion of why we didn't do this

**Review by ararslan — APPROVED** [review]

**Comment by kshyatt** [comment]
Test fail is in unrelated spawn tests.

**Pull request merged**